src/Microsoft.AspNetCore.OData/Extensions/ODataRouteBuilderExtensions.cs

```diff
@@ -39,23 +38,7 @@ public static IRouteBuilder SetDefaultQuerySettings(this IRouteBuilder builder,
                 throw Error.ArgumentNull("builder");
             }
 
-            if (defaultQuerySettings == null)
-            {
-                throw Error.ArgumentNull("defaultQuerySettings");
-            }
-
-            if (!defaultQuerySettings.MaxTop.HasValue || defaultQuerySettings.MaxTop > 0)
-            {
-                ModelBoundQuerySettings.DefaultModelBoundQuerySettings.MaxTop = defaultQuerySettings.MaxTop;
-            }
-
-            DefaultQuerySettings querySettings = builder.ServiceProvider.GetRequiredService<DefaultQuerySettings>();
-            if (querySettings == null)
-            {
-                throw Error.InvalidOperation(SRResources.MissingODataServices, nameof(DefaultQuerySettings));
-            }
-
-            querySettings = defaultQuerySettings;
+            builder.ServiceProvider.SetDefaultQuerySettings(defaultQuerySettings);
             return builder;
```

Comment:
What is the reason behind this change? Extension method for service provider is not intuitive. May be you want to create a private static methods rather than extension methods.

Reply:
Synced offline, right?

---
In reply to: [375976380](https://github.com/OData/WebApi/pull/2035#discussion_r375976380) [](ancestors = 375976380)